AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -23,9 +33,67 @@ class DoubleVideoViewController: UIViewController, CellTitled {
 
     override func viewDidLayoutSubviews() {
         // update your layers' frames here
+        guard let sublayersForTop = self.videoContainerTop.layer.sublayers else { return }
+        guard let sublayersForBottom = self.videoContainerBottom.layer.sublayers else { return }
+        
+        for layers in sublayersForTop {
+            layers.frame = self.videoContainerTop.bounds
+        }
+        for layers in sublayersForBottom {
+            layers.frame = self.videoContainerBottom.bounds
+        }
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        let picker = UIImagePickerController()
+        picker.sourceType = .photoLibrary
+        picker.delegate = self
+        picker.mediaTypes = [String(kUTTypeImage), String(kUTTypeMovie)]
+        present(picker, animated: true)
+    }
     
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        switch info[UIImagePickerControllerMediaType] as! String {
+        case String(kUTTypeImage):
+            break
+        case String(kUTTypeMovie):
+            if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+                if self.firstVideoURL == nil {
+                    self.firstVideoURL = url
```

Comment:
Checking the rate of the player was specified and would be a better strategy for determining slot availability. Also, because you don't ```nil``` the URL variables these blocks can't be entered after those vars are set.